src/gwt/src/org/rstudio/studio/client/workbench/views/jobs/view/JobProgress.java

```diff
@@ -63,26 +76,33 @@ public void showJob(Job job)
       name_.setText(job.name);
       progress_.setLabel(job.name);
       String status = JobConstants.stateDescription(job.state);
+      stop_.addClickHandler(clickEvent ->
+         eventBus_.fireEvent(new JobExecuteActionEvent(job.id, JobConstants.ACTION_STOP)));
+
       if (job.completed > 0)
       {
          // Job is not running; show its completion status and time
          progress_.setVisible(false);
          status += " " + StringUtil.friendlyDateTime(new Date(job.completed * 1000));
          elapsed_.setText(StringUtil.conciseElaspedTime(job.completed - job.started));
          status_.setVisible(true);
+         stop_.setVisible(false);
       }
       else if (job.max > 0)
       {
-         // Job is running and has a progress bar; show it and hide the status indicator
+         // Job is running and has a progress bar;
+         // show it with stop button and hide the status indicator
          progress_.setVisible(true);
          progress_.setProgress(job.progress, job.max);
          status_.setVisible(false);
+         stop_.setVisible(true);
```

Comment:
Whether or not a job is stoppable is independent of whether or not it has a progress bar. I think we should make the button visible based on whether or not the job implements the `stop` action.

Reply:
Thanks for pointing this out! I reverted the original comments and updated the stop button logic.